index.js

```diff
@@ -6,15 +6,19 @@ function ghGot(path, opts) {
 		return Promise.reject(new TypeError(`Expected 'path' to be a string, got ${typeof path}`));
 	}
 
-	opts = Object.assign({json: true, endpoint: 'https://api.github.com/'}, opts);
+	const env = process.env;
+
+	opts = Object.assign({
+		json: true,
+		endpoint: env.GITHUB_ENDPOINT ? env.GITHUB_ENDPOINT.replace(/[^/]$/, '$&/') : 'https://api.github.com/'
+	}, opts);
 
 	opts.headers = Object.assign({
 		'accept': 'application/vnd.github.v3+json',
 		'user-agent': 'https://github.com/sindresorhus/gh-got'
 	}, opts.headers);
 
-	const env = process.env;
-	const token = env.GITHUB_TOKEN || opts.token;
+	const token = opts.token || env.GITHUB_TOKEN;
```

Comment:
:+1:

I guess it's no big deal but maybe we should remove `opts.endpoint` and `opts.token` before sending them to `got`.